lib/importer/nci_thesaurus_mirror.rb

```diff
@@ -0,0 +1,63 @@
+module Importer
+  class NciThesaurusMirror
+    attr_reader :parser, :version
+
+    def initialize(path, version = Time.now.utc.iso8601)
+      @parser = Obo::Parser.new(path)
+      @version = version
+    end
+
+    def import
+      parser.elements.each do |elem|
+        if valid_entry?(elem)
+          create_object_from_entry(elem)
+        end
+      end
+    end
+
+    def valid_entry?(entry)
+      semantic_types = semantic_types(entry)
+      (entry['id'].present? && entry['name'].present? && entry.respond_to?(:name) && entry.name == 'Term' &&
+        (semantic_types & ['Pharmacologic Substance', 'Pharmacological Substance', 'Clinical Drug', 'Therapeutic or Preventive Procedure']).length > 0 &&
+        (semantic_types & ['Obsolete_Concept']).length == 0)
+    end
+
+    def semantic_types(entry)
+        entry['property_value'].select{|s| s.start_with? 'NCIT:P106'}.map{|t| t.rpartition(' ')[0].split(' ', 2)[1].gsub!('"', '')}
+    end
+
+    def create_object_from_entry(entry)
+      name = Drug.capitalize_name(entry['name'])
+      ncit_id = entry['id'].sub('NCIT:', '')
+      drug = ::Drug.where(ncit_id: ncit_id).first_or_initialize
+      drug.name = name
+      synonyms = process_synonyms(entry['synonym']).uniq
+      synonyms.each do |syn|
+        drug_alias = ::DrugAlias.where(name: syn).first_or_create
+        if not drug.drug_aliases.map{|a| a.name.downcase}.include? syn.downcase and not syn.downcase.eql? drug.name.downcase
```

Comment:
Generally speaking you'll want to use `!` instead of `not` and `==` instead of `.eql?` for this sort of thing. `not` has a weird and different operator precedence than you might expect and can sometimes behave in ways that are odd.